ecs-cli/modules/clients/aws/ecr/client.go

```diff
@@ -65,6 +66,43 @@ func NewClient(config *config.CommandConfig) Client {
 	return newClient(config, client, loginClient)
 }
 
+// NewFipsClient Creates a new ECR client that will communicate with a FIPS endpoint.
+func NewFipsClient(config *config.CommandConfig) (Client, error) {
+	region := aws.StringValue(config.Session.Config.Region)
+	resolver := endpoints.DefaultResolver()
+
+	// The convention for FIPS endpoints is to add "-fips" to the official
+	// service name; however this would result in an UnknownServiceError
+	// being thrown. To allow the endpoint resolver to fallback to the
+	// correct endpoint, we can set the ResolveUnknownService flag to true.
+	endpoint, err := resolver.EndpointFor("ecr-fips", region, func(opts *endpoints.Options) {
+		opts.ResolveUnknownService = true
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Using FIPS endpoint: %+v", endpoint.URL)
+
+	awsSession := config.Session
+	awsConfig := awsSession.Config.WithEndpoint(endpoint.URL)
```

Comment:
Just noting that because of pointers, this updates the endpoint in the original config. Thankfully, this isn't a problem in the ECR commands because this is the only client we use. However, it might be good to put a comment here for future reference.

See: https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config.WithEndpoint